contracts/libraries/HyperdriveExecution.sol

```diff
@@ -0,0 +1,578 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.20;
+
+import { IHyperdrive } from "hyperdrive/contracts/src/interfaces/IHyperdrive.sol";
+import { FixedPointMath } from "hyperdrive/contracts/src/libraries/FixedPointMath.sol";
+import { HyperdriveMath } from "hyperdrive/contracts/src/libraries/HyperdriveMath.sol";
+import { SafeCast } from "hyperdrive/contracts/src/libraries/SafeCast.sol";
+import { YieldSpaceMath } from "hyperdrive/contracts/src/libraries/YieldSpaceMath.sol";
+import { Packing } from "openzeppelin/utils/Packing.sol";
+import { IEverlong } from "../interfaces/IEverlong.sol";
+import { IEverlongEvents } from "../interfaces/IEverlongEvents.sol";
+
+// TODO: Extract into its own library.
+uint256 constant HYPERDRIVE_SHARE_RESERVES_BOND_RESERVES_SLOT = 2;
+uint256 constant HYPERDRIVE_LONG_EXPOSURE_LONGS_OUTSTANDING_SLOT = 3;
+uint256 constant HYPERDRIVE_SHARE_ADJUSTMENT_SHORTS_OUTSTANDING_SLOT = 4;
+
+/// @author DELV
+/// @title HyperdriveExecutionLibrary
+/// @notice Library to handle the execution of trades with hyperdrive.
+/// @custom:disclaimer The language used in this code is for coding convenience
+///                    only, and is not intended to, and does not, have any
+///                    particular legal or regulatory significance.
+library HyperdriveExecutionLibrary {
+    using FixedPointMath for uint256;
+    using SafeCast for *;
+    using Packing for bytes32;
+
+    // ╭─────────────────────────────────────────────────────────╮
+    // │ Open Long                                               │
+    // ╰─────────────────────────────────────────────────────────╯
+
+    /// @notice Opens a long with hyperdrive using amount.
+    /// @param _asBase Whether to use hyperdrive's base asset.
+    /// @param _amount Amount of assets to spend.
+    /// @return maturityTime Maturity timestamp of the opened position.
+    /// @return bondAmount Amount of bonds received.
+    function openLong(
```

Comment:
It would be good to add an extra data field to this for future proofing. We currently don't use this, but for yield sources that require some kind of KYC or off-chain attestation (some examples of this are RWA protocols like Ondo), will require extra data to be passed.